### ISSUE TITLE
Terraform Azure Provider Crash during deploy (on Azure Function Host)

### ISSUE THREAD

**Issue by makaanneo** [hashicorp/terraform-provider-azurerm#6773]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```powershell
Terraform v0.12.24
+ provider.azurerm v2.8.0
```

### Affected Resource(s)

* `azurerm_resource_group`

### Terraform Configuration Files

```hcl
locals {
  suffix          = "makaanneo"
}

provider "azurerm" {
  features {}

  subscription_id = var.subscription_id
  client_id       = var.client_id
  client_secret   = var.client_secret
  tenant_id       = var.tenant_id
}

resource "random_string" "random" {
  length  = 15
  special = false
  lower   = true
  upper   = false
}

variable "tenant_id" {}
variable "subscription_id" {}
variable "client_id" {}
variable "client_secret" {}

resource "azurerm_resource_group" "example" {
  name     = "${random_string.random.result}rg${local.suffix}"
  location = "Germany West Central"
}
```

### Debug Output

https://gist.github.com/makaanneo/6cc54502c1eda40b9075c4d491fbf7aa

### Panic Output

https://gist.github.com/makaanneo/6cc54502c1eda40b9075c4d491fbf7aa

### Expected Behavior

no crash

### Actual Behavior

crash during deployment

### Steps to Reproduce

Prepare environment on Azure
1. Create function app version 3
2. Windows X64 Service Plan
3. Open Deployment FTP
4. Upload test script via FTP

#### Steps
1. `terraform init`
1. `terraform plan` or 
1. `terraform apply`

### Important Factoids

Terraform is used in a Azure Function Host environment to deploy/modify resources in other Subscriptions on Azure.

Used Region *Germany West Central*

Identity on function app is switched on to use *SystemAssigned*

### References

* #0000

**Comment by artemius84** [comment]
👍

**Comment by ArcturusZhang** [comment]
Hi @makaanneo thanks for this issue and sorry for the inconvenience.

Since it is not quite applicable for me to create exactly the same environment as yours to reproduce this issue, I just briefly read your panic output. The panic comes from go-autorest/adal, it seems something is wrong during the authorization.
I will take this issue to someone to have a look.

**Comment by makaanneo** [comment]
Hi @ArcturusZhang thanks

**Comment by ArcturusZhang** [comment]
Hi @makaanneo could you provide more context about this?
```
Terraform is used in a Azure Function Host environment to deploy/modify resources in other Subscriptions on Azure.
```
I believe this is the key point for reproducing this issue (currently I cannot reproduce this even when using the same location)

**Comment by ArcturusZhang** [comment]
Hi @makaanneo we are still investigating this issue but still cannot get the root cause of this...

Could you please help us a bit for further investigation?

1. Does this issue happen on occasion or all the time?
1. Could you please try the latest azurerm provider to see if it solves your issue? Based on your panic log, your terraform provider is running on `go1.12.13`, since this panic comes out of some HTTP communication, this may be also an important factoid.
1. Could you please also enable the HTTP log in the SDK by exporting `AZURE_GO_SDK_LOG_LEVEL=DEBUG` and `AZURE_GO_SDK_LOG_FILE=/path/to/your-sdk-log-file`?

**Comment by jhendrixMSFT** [comment]
This is fixed in autorest/adal/v0.9.2

**Comment by favoretti** [comment]
Since this issue has been reported a long time ago and relates to the version of provider we no longer support - I'm going to close it. Please open a new updated bug report on current versions of terraform and provider if this is still relevant. Thank you.
